### PR TITLE
Network

### PR DESCRIPTION
**Pull request by Benau**

### CODE REVIEW

**Comment by Benau** [comment]
the important changes are in *kart animtion (it's redone so the item can be rewound with maunal function)
1ac4a5b, cfbbc7c, d00472f, 2aede6f

attachement* (swatter)
c57af03, 17fd68e, 9b8b0b7, 1193c58

check_trigger (convert trigger item to check structure, so all itemstate can be graphically corrected)
fe4a9c5, 9b4b283

max_speed optimization
c00cf66

**Comment by deveee** [comment]
Atm. github shows merge conflict and also probably protocol version should be updated?

I was testing it a bit and I didn't find any issues. I will do more tests in next days. @hiker you know this code better. Could you have a look?

**Comment by hiker** [comment]
Yes, I have already started, but some of the changes will take some time :( And yes, I totally agree with protocol number.

File: src/utils/mini_glm.hpp [hunk 1]
```diff
@@ -565,6 +568,42 @@ namespace MiniGLM
...
     // ------------------------------------------------------------------------
+    inline void compressbtTransform(btTransform& cur_t,
+                                    int* compressed_data = NULL)
+    {
+        int x = (int)(cur_t.getOrigin().x() * 100.0f);
+        int y = (int)(cur_t.getOrigin().y() * 100.0f);
+        int z = (int)(cur_t.getOrigin().z() * 100.0f);
+        x = core::clamp(x, -8388608, 8388607);
+        y = core::clamp(y, -8388608, 8388607);
+        z = core::clamp(z, -8388608, 8388607);
```
Comment (hiker) [inline src/utils/mini_glm.hpp]: Hex numbers, and a description of what the function does.

File: src/karts/moveable.cpp [hunk 2]
```diff
@@ -33,8 +33,6 @@
 
 Moveable::Moveable()
 {
-    m_body            = 0;
```
[the remark this answers is not in the document]
Comment (Benau) [reply src/karts/moveable.cpp]: Std unique ptr

**Comment by hiker** [comment]
This pull request contains a lot of useful improvements, thanks a lot. The way it is tbh it's more or less impossible to properly review - you can't go by commit (since then you miss later changes), and you can't review it all since it's way too much. I've done a review and provided some feedback. But giving our timing constraints I don't want to waste more time by asking this to be split into its components 📦 
The conflict needs to be resolved, network version number be increased, and maybe @Alayan-stk-2 can confirm that it works as expected and then merge it in, so we can get started on our rc?

**Comment by Benau** [comment]
Done

Now i suggest you wait 1-2 days for samuncle to fix his gloss map.....

**Comment by leyyin** [comment]
Usually when someone creates a PR even if they have write access to that repository, it is expected of other team members to review and merge that PR.

**Comment by Arthur-D** [comment]
Proper process and best practices are mostly ignored in this project, and have even been vocally derided (oh no, a branch everyone can't push to without review, such horrible acts of despotism!), so at this point there isn't much use in trying to bring them back to the discussion.

**Pull request closed without merging**